lisa/operating_system.py

```diff
@@ -504,26 +500,40 @@ def _package_exists(self, package: str, signed: bool = True) -> bool:
         return False
 
     def _get_information(self) -> OsInformation:
-        cmd_result = self._node.execute(
-            cmd="lsb_release -a", shell=True, no_error_log=True
-        )
+        # try to set version info from /etc/os-release.
+        cat = self._node.tools[Cat]
+        cmd_result = cat.run("/etc/os-release")
         cmd_result.assert_exit_code(message="error on get os information")
-        assert cmd_result.stdout, "not found os information from 'lsb_release -a'"
 
+        vendor: str = ""
+        release: str = ""
+        codename: str = ""
+        full_version: str = ""
         for row in cmd_result.stdout.splitlines():
-            os_release_info = self.__lsb_os_info_pattern.match(row)
-            if os_release_info:
-                if os_release_info.group("name") == "Distributor ID":
-                    vendor = os_release_info.group("value")
-                elif os_release_info.group("name") == "Release":
-                    release = os_release_info.group("value")
-                elif os_release_info.group("name") == "Codename":
-                    codename = os_release_info.group("value")
-                elif os_release_info.group("name") == "Description":
-                    full_version = os_release_info.group("value")
+            os_release_info = super()._os_info_pattern.match(row)
+            if not os_release_info:
+                continue
+            if os_release_info.group("name") == "NAME":
+                vendor = os_release_info.group("value")
+            elif os_release_info.group("name") == "VERSION":
+                codename = get_matched_str(
+                    os_release_info.group("value"),
+                    super()._distro_codename_pattern,
+                )
+            elif os_release_info.group("name") == "PRETTY_NAME":
+                full_version = os_release_info.group("value")
+
+        # get release from /etc/debian_version for debian distro
+        # credativ debian 9-backports 9.20190313.0
```

Comment:
Add more explanation, so people know "credativ debian 9-backports 9.20190313.0" is a marketplace image name, and what's the raw output line.

Reply:
update